src/solver/variable/include/antares/solver/variable/economy/minDispatchableGenByPlant.h

```diff
@@ -0,0 +1,281 @@
+                                            { return c->isEnabled() && !c->isMustRun(); });
+        if (nbClusters_)
+        {
+            AncestorType::pResults.resize(nbClusters_);
+
+            for (unsigned int numSpace = 0; numSpace < pNbYearsParallel; numSpace++)
+            {
+                pValuesForTheCurrentYear[numSpace].resize(nbClusters_);
+            }
+
+            for (unsigned int numSpace = 0; numSpace < pNbYearsParallel; numSpace++)
+            {
+                for (unsigned int i = 0; i != nbClusters_; ++i)
+                {
+                    pValuesForTheCurrentYear[numSpace][i].initializeFromStudy(*study);
+                }
+            }
+
+            for (unsigned int i = 0; i != nbClusters_; ++i)
+            {
+                AncestorType::pResults[i].initializeFromStudy(*study);
+                AncestorType::pResults[i].reset();
+            }
+        }
+        else
+        {
+            AncestorType::pResults.clear();
+        }
+        // Next
+        NextType::initializeFromArea(study, area);
+    }
+
+    size_t getMaxNumberColumns() const
+    {
+        return nbClusters_ * ResultsType::count;
+    }
+
+    void yearBegin(unsigned int year, unsigned int numSpace)
+    {
+        // Reset the values for the current year
+        for (unsigned int i = 0; i != nbClusters_; ++i)
+        {
+            pValuesForTheCurrentYear[numSpace][i].reset();
+        }
+        // Next variable
+        NextType::yearBegin(year, numSpace);
+    }
+
+    void yearEnd(unsigned int year, unsigned int numSpace)
+    {
+        // Merge all results for all thermal clusters
+        {
+            for (unsigned int i = 0; i < nbClusters_; ++i)
+            {
+                // Compute all statistics for the current year (daily,weekly,monthly)
+                pValuesForTheCurrentYear[numSpace][i].computeStatisticsForTheCurrentYear();
+            }
+        }
+        // Next variable
+        NextType::yearEnd(year, numSpace);
+    }
+
+    void computeSummary(std::map<unsigned int, unsigned int>& numSpaceToYear,
+                        unsigned int nbYearsForCurrentSummary)
+    {
+        for (unsigned int numSpace = 0; numSpace < nbYearsForCurrentSummary; ++numSpace)
+        {
+            for (unsigned int i = 0; i < nbClusters_; ++i)
+            {
+                // Merge all those values with the global results
+                AncestorType::pResults[i].merge(numSpaceToYear[numSpace],
+                                                pValuesForTheCurrentYear[numSpace][i]);
+            }
+        }
+
+        // Next variable
+        NextType::computeSummary(numSpaceToYear, nbYearsForCurrentSummary);
+    }
+
+    void hourForEachArea(State& state, unsigned int numSpace)
+    {
+        auto& area = state.area;
+        auto& thermal = state.thermal;
+        unsigned indexCluster = 0;
+        for (auto& cluster: area->thermal.list.each_enabled_and_not_mustrun())
```

Comment:
As discussed, please be consistent with other "by plant" variables

=> use `thermal.list.each_enabled()`

Reply:
Done